src/fraction.ts

```diff
@@ -0,0 +1,69 @@
+import BN from "bn.js";
+
+export class Fraction {
+  private numerator: BN;
+  private denominator: BN;
+
+  constructor(numerator: BN | number, denominator: BN | number) {
+    this.numerator = new BN(numerator);
+    this.denominator = new BN(denominator);
+
+    if (this.denominator.isZero()) {
+      throw Error("Denominator cannot be zero");
+    }
+  }
+
+  gt(other: Fraction) {
+    return this.sub(other).toNumber() > 0;
```

Comment:
I would rather avoid relying on floats since we are using BN anyway. Another way to rephrase this comparison using only `BN` could be:
```
return this.numerator.mul(other.denominator).gt(other.numerator.mul(this.denominator))
```

Reply:
Sorry, this is actually a bad recommendation since the denominators could be negative.
A working comparison is:
```
const diff = this.sub(other)
return diff.numerator.mul(diff.denominator).gt(new BN(0))
```
You multiply numerator and denominator of the difference. If the sign is positive, then the difference is positive, otherwise it's not.